src/refmvs.rs

```diff
@@ -1696,14 +1698,18 @@ pub(crate) fn rav1d_refmvs_init_frame(
                 frm_hdr.frame_offset as i32,
             );
             if diff1.abs() > 31 {
-                rf.mfmv_ref2cur[n] = i32::MIN;
+                rf.mfmv_ref2cur[n] = INVALID_REF2CUR;
             } else {
-                rf.mfmv_ref2cur[n] = if rf.mfmv_ref[n] < 4 { -diff1 } else { diff1 };
+                rf.mfmv_ref2cur[n] = if rf.mfmv_ref[n] < 4 {
+                    -diff1 as i8
+                } else {
+                    diff1 as i8
+                };
                 for m in 0..7 {
                     let rrpoc = ref_ref_poc[rf.mfmv_ref[n] as usize][m];
                     let diff2 = get_poc_diff(seq_hdr.order_hint_n_bits, rpoc as i32, rrpoc as i32);
                     // unsigned comparison also catches the < 0 case
-                    rf.mfmv_ref2ref[n][m] = if diff2 as u32 > 31 { 0 } else { diff2 };
+                    rf.mfmv_ref2ref[n][m] = if diff2 as u32 > 31 { 0 } else { diff2 as u8 };
```

Comment:
Is the `diff as u32 > 31` still correct (or idiomatic) with the new smaller types?